utils/crypto_utils.h

```diff
@@ -20,16 +20,90 @@ extern "C" {
 #define MAXLINE 1024
 #define IMSI_LEN 15
 
-int get_device_id(const char *device_id);
-int get_aes_key(const uint8_t *key);
+/**
+ * @brief Gets device IMSI or physical ID
+ *
+ * @param[out] device_id Device id output
+ *
+ * @return #retcode_t
+ */
+retcode_t get_device_id(const char *device_id);
+
+/**
+ * @brief Gets aes key from hashchain
+ *
+ * @param[out] key Device key output
+ *
+ * @return #retcode_t
+ */
+retcode_t get_aes_key(const uint8_t *key);
+
+/**
+ * @brief Encrypt plaintext to ciphertext
+ *
+ * @param[in] plaintext The text to be encrypted
+ * @param[in] plaintext_len Plaintext length
+ * @param[out] ciphertext Ciphrtext
+ * @param[in] ciphertext_len Ciphertext length
+ * @param[in, out] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[in] key[AES_CBC_KEY_SIZE] Encryption key
+ * @param[in] device_id[IMSI_LEN + 1] Device id
+ *
+ * @return
+ * - Ciphertext text new length on success
+ * - 0 on error
+ */
 int encrypt(const char *plaintext, int plaintext_len, char *ciphertext, int ciphertext_len, uint8_t iv[AES_BLOCK_SIZE],
             uint8_t key[AES_CBC_KEY_SIZE], uint8_t device_id[IMSI_LEN + 1]);
-int decrypt(const char *ciphertext, int ciphertext_len, char *plaintext, int plaintext_len, uint8_t iv[AES_BLOCK_SIZE],
-            uint8_t key[AES_CBC_KEY_SIZE]);
+
+/**
+ * @brief Decrypt ciphertext to plaintext
```

Comment:
Simply say, "Decrypt ciphertext"

Reply:
It is not necessary to mention "to plaintext" since "decrypt" always implies that.